src/lib/db/transaction.ts

```diff
@@ -86,6 +93,24 @@ export function withTransactional<S>(
     return service;
 }
 
+export function withContextualTransactional<S>(
+    serviceFactory: (db: Knex) => S,
+    db: Knex,
+): withContextualTransactional<S> {
+    const service = serviceFactory(db) as withContextualTransactional<S>;
+
+    service.trackedTransactional = async <R>(
+        fn: (transactionContext: TransactionContext) => R,
+    ): Promise<R> => {
+        return db.transaction(async (trx) => {
+            const transactionContext = new TransactionContext(trx);
+            return fn(transactionContext);
```

Comment:
I'd not be following this... `fn` would be the original function from the service, but here you're changing it's method signature to accept a `TransactionContext`, is that correct?